### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by rinckm**
### Description

- Introduce constants where strings are used multiple times
- Remove unnecessary boolean literals
- Use opposite operator in favour of negation
- Reduce complexity of functions
- Change function names to better reflect their semantics

### Submitter checklist

- [x] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: pkg/runctl/secretmgr/secret_manager.go [hunk 1]
```diff
@@ -12,6 +12,11 @@ import (
 	klog "k8s.io/klog/v2"
 )
 
+const (
+	jenkinsIO = "jenkins.io/"
+	tektonDev = "tekton.dev/"
```
Comment (anbrsap) [inline pkg/runctl/secretmgr/secret_manager.go:17]:
```suggestion
	annotationPrefixJenkins = "jenkins.io/"
	annotationPrefixTekton = "tekton.dev/"
```

File: pkg/k8s/pipelineRun.go [hunk 2]
```diff
@@ -380,28 +385,44 @@ func (r *pipelineRun) CommitStatus(ctx context.Context) ([]*api.StateItem, error
...
+	for _, recorder := range r.commitRecorders {
+		if recorder != nil {
+			result = append(result, recorder())
+		}
+	}
+	return result, errors.Wrapf(err, "failed to update status [%s]", r.String())
+}
 
-		if retryCount > 0 {
+func (r *pipelineRun) doRetry(ctx context.Context, retryState *retryData) func() error {
```
[the remark this answers is not in the document]
Comment (rinckm) [reply pkg/k8s/pipelineRun.go:422]: The point is that we need to reduce the cognitive complexity.

File: changelog.yaml [hunk 3]
```diff
@@ -51,6 +51,16 @@
 - version: NEXT
   date: TBD
   changes:
+    - type: internal
+      impact: minor
```
[the remark this answers is not in the document]
Comment (rinckm) [reply changelog.yaml]: done

File: pkg/runctl/run_manager.go [hunk 4]
```diff
@@ -239,6 +218,36 @@ func (c *runManager) setupServiceAccount(ctx context.Context, runCtx *runContext
 	return nil
 }
 
+func attachAllSecrets(ctx context.Context, runCtx *runContext, accountManager k8s.ServiceAccountManager, serviceAccount *k8s.ServiceAccountWrap, pipelineCloneSecretName string, imagePullSecrets []string) error {
```
[the remark this answers is not in the document]
Comment (rinckm) [reply pkg/runctl/run_manager.go]: Done

File: pkg/runctl/runmgr/run_manager.go [hunk 5]
```diff
@@ -239,6 +215,36 @@ func (c *runManager) setupServiceAccount(ctx context.Context, runCtx *runContext
 	return nil
 }
 
+func (c *runManager) attachAllSecrets(ctx context.Context, runCtx *runContext, accountManager k8s.ServiceAccountManager, serviceAccount *k8s.ServiceAccountWrap, pipelineCloneSecretName string, imagePullSecrets []string) error {
```
Comment (anbrsap) [inline pkg/runctl/runmgr/run_manager.go:218]: Param `serviceAccount` is not needed.

Comment (rinckm) [reply pkg/runctl/runmgr/run_manager.go:218]: removed

**Review by anbrsap — APPROVED** [review]

**Pull request merged**